src/HexagonOffload.cpp

```diff
@@ -318,8 +318,11 @@ void do_reloc(char *addr, uint32_t mask, uintptr_t val, bool is_signed, bool ver
         } else if ((inst >> 24) == 65 || (inst >> 24) == 77) {
             // 0100 0001 001s ssss PP0t tiii iiid dddd
             mask = 0x000007e0;
+        } else if ((inst >> 21) == 540) {
```

Comment:
@abadams @pranavb-ca can you check this? I checked exactly all of the bits that are 0 or 1 in the instruction, which doesn't seem to be the case in the above checks (e.g. the ones on line 318 don't check the last 3 bits of 0 or 1). I think either my new check is wrong, or the ones on line 318 are wrong (and maybe more, didn't look).

Reply:
Or, maybe the problem is just that the one on line 318 is missing all of the instruction encodings it is meant to support? At the very least it should have two, maybe it really needs a large list, where any value for the 3 unchecked bits uses the same mask?

Reply:
Just in trying to read this, wouldn't it be better to use hexadecimal constants in the if statements?

Reply:
I agree, but I decided to follow the convention set by the other constants here for now rather than introduce a mix of hex and decimal constants, and avoid making extra changes in this PR.

Reply:
Hmm, I realized this isn't hard to check on myself. It seems like the check for == 65 makes sense, because there are a bunch of instructions with the same mask that that check catches. I added the instructions to the comment.

I can't find the == 77 case in the instruction encodings. That one seems like it might be incorrect? @pranavb-ca you added both of these in https://github.com/halide/Halide/commit/4c2b08f6f3914916b5b6de37ae7d7eae81faeaa7#diff-29c2b5f56ba040c9426b814755869e62, can you add a comment for the instructions the 77 mask is meant to check?

Reply:
I spent some time staring at these things and decided that the right way to fix this was to just automate away this stuff: https://github.com/halide/Halide/compare/move-to-runtime...automate-hex-instructions?expand=1 I'm going to leave that in a separate branch for now, I'll open a PR later.